CefSharp.BrowserSubprocess.Core/CefAppUnmanagedWrapper.cpp

```diff
@@ -149,50 +160,88 @@ namespace CefSharp
-                auto postMessageFunction = CefV8Value::CreateFunction(kPostMessage, new JavascriptPostMessageHandler(rootObject == nullptr ? nullptr : rootObject->CallbackRegistry));
-                auto promiseHandlerFunction = CefV8Value::CreateFunction(kSendEvalScriptResponse, new JavascriptPromiseHandler());
-
-                //By default We'll support both CefSharp and cefSharp, for those who prefer the JS style
-                auto createCefSharpObj = !_jsBindingPropertyName.empty();
-                auto createCefSharpObjCamelCase = !_jsBindingPropertyNameCamelCase.empty();
-
-                if (createCefSharpObj)
+                auto createObjects = true;
+
+                if (_jsBindingApiHasAllowOrigins)
                 {
-                    auto cefSharpObj = CefV8Value::CreateObject(nullptr, nullptr);
-                    cefSharpObj->SetValue(kBindObjectAsync, bindObjAsyncFunction, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-                    cefSharpObj->SetValue(kDeleteBoundObject, unBindObjFunction, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-                    cefSharpObj->SetValue(kRemoveObjectFromCache, removeObjectFromCacheFunction, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-                    cefSharpObj->SetValue(kIsObjectCached, isObjectCachedFunction, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-                    cefSharpObj->SetValue(kPostMessage, postMessageFunction, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-                    cefSharpObj->SetValue(kSendEvalScriptResponse, promiseHandlerFunction, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-                    cefSharpObj->SetValue(kRenderProcessId, CefV8Value::CreateInt(processId), CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_NONE);
-
-                    global->SetValue(_jsBindingPropertyName, cefSharpObj, CefV8Value::PropertyAttribute::V8_PROPERTY_ATTRIBUTE_READONLY);
+                    createObjects = false;
+
+                    auto frameUrl = frame->GetURL();
+
+                    CefURLParts frameUrlParts;
+
+                    if (CefParseURL(frameUrl, frameUrlParts))
+                    {
+                        auto frameUrlOrigin = CefString(frameUrlParts.origin.str, frameUrlParts.origin.length);
+                        auto clrframeUrlOrigin = StringUtils::ToClr(frameUrlOrigin);
+
+                        auto size = static_cast<int>(_jsBindingApiAllowOrigins->GetSize());
+
+                        for (int i = 0; i < size; i++)
+                        {
+                            auto origin = _jsBindingApiAllowOrigins->GetString(i);
+
+                            auto clrOrigin = StringUtils::ToClr(origin);
+
+                            auto originEqual = String::Compare(clrframeUrlOrigin, clrOrigin, StringComparison::InvariantCultureIgnoreCase);
```

Comment:
Need to find a more efficient string comparison, ideally compare the native strings